Makefile

```diff
@@ -38,12 +38,30 @@ lint: lint.sub
 push: push.sub
 test: test.sub
 
+clean:
+	rm -f .dinit
+	rm -f .buildImage
+	docker rmi -f buildImage > /dev/null 2>&1 || echo -n ""
+
+docker-all: .buildImage
+	docker run --rm -ti \
+	  -v $(PWD):/go/src/github.com/kubernetes-incubator/service-catalog \
+	  buildimage make .dinit all
+
+.dinit: glide.yaml
```

Comment:
So, here's my dirty secret: I suck at reading makefiles.  I'm totally cool with learning, but in the meantime it is sort of hard for me to distinguish what is some construct of make and what is naming.  Does the `d` in `.dinit` stand for dependency?  Why the leading `.`?

Anyway, I think a comment for each make directive is very desirable.